main/src/library/Option.flix

```diff
@@ -185,19 +178,11 @@ namespace Option {
     }
 
     ///
-    /// Returns `Ok(v)` if `o` is `Some(v)`. Otherwise returns `Err(z)`.
-    ///
-    def toOk[t, e](o: Option[t], z: e): Result[t, e] = match o with {
-        case None => Err(z)
-        case Some(v) => Ok(v)
-    }
-
-    ///
-    /// Returns `Err(v)` if `o` is `Some(v)`. Otherwise returns `Ok(z)`.
+    /// Returns a singleton map with the mapping `k -> v` if `o` is `Some((k, v))`. Otherwise returns the empty map.
     ///
-    def toErr[t,e](o: Option[e], z: t): Result[t,e] = match o with {
-        case None => Ok(z)
-        case Some(v) => Err(v)
+    def toMap[k,v](o: Option[(k, v)]): Map[k, v] = match o with {
+        case None => @{}
+        case Some((k, v)) => Map.singleton(k, v)
```

Comment:
If I use `@{k -> v}` instead of `Map.singleton(k, v)`, the following error occurs:  
`An exception or error caused a run to abort: Attempt to retrieve value from Failure. The errors are: ParseError(Invalid input '}', expected LegalLetter, "->", '.', '(' or optWS (line 185, column 38):
        case Some((k, v)) => @{k -> v}
                                                          ^,Internal(Option.flix,/*`
I can look into this later if it is incorrect behaviour.